### ISSUE TITLE
SigV4Auth.auth argument `request` hinted as incorrect type `requests.Request`

### ISSUE THREAD

**Issue by mk-tng** [vemel/mypy_boto3_builder#109]
**Describe the bug**
`SigV4Auth.auth()`  in https://github.com/vemel/mypy_boto3_builder/blob/master/mypy_boto3_builder/botocore_stubs_static/auth.pyi incorrectly hints `request` as `requests.Request` instead of `botocore.awsrequest.AWSRequest`. At runtime, using an instance of`requests.Request` raises an error.

While `botocore.awsrequest.AWSRequest` is largely based on `requests.Request` (as stated in the class comment), both have diverged and are not fully compatible.
(`request.Request` is used several times in the module and in botocore_stubs_static. Its not directly clear where it will work or not.)

**To Reproduce**
1. pip install boto3 boto3-stubs mypy requests types-requests
2. Run mypy on the following code sample

```python
from boto3 import Session
from botocore.auth import SigV4Auth
from requests import Request


def main() -> None:
    boto_session = Session()
    request = Request(url="https://some.example")
    sigv4 = SigV4Auth(boto_session.get_credentials(), "execute-api", "eu-central-1")
    sigv4.add_auth(request)  # fails


if __name__ == "__main__":
    main()
```
3. Actual mypy output
```
Success: no issues found in 1 source file
```

**Expected mypy-output**

An error that `request` is of incorrect type. The correct type for argument `request` in `SigV4Auth.auth` should be `botocore.awsrequest.AWSRequest`.

4. Invoking the example:
```
Traceback (most recent call last):
  File "example.py", line 14, in <module>
    main()
  File "example.py", line 10, in main
    sigv4.add_auth(request)  # fails
  File "/home/kilgerm/.pyenv/versions/tmp1/lib/python3.8/site-packages/botocore/auth.py", line 375, in add_auth
    request.context['timestamp'] = datetime_now.strftime(SIGV4_TIMESTAMP)
AttributeError: 'Request' object has no attribute 'context'
```

**Additional context**
Ubuntu 20.04
Installed via pip
boto3==1.18.43
boto3-stubs==1.18.43
botocore==1.21.43
botocore-stubs==1.21.43

**Comment by vemel** [comment]
Thanks for the bug report!

I prepared a fix, going to release it today so it will be included in tomorrow `boto3-stubs 1.18.44` release.

**Comment by mk-tng** [comment]
Thank you for the quick fix! :heart:
